### PR TITLE
add few options for TI analysis and update the README file

### PR DESCRIPTION
**Pull request by shuail**
I add few options for TI analysis 1, specific the portion of the energy files to be analysis for TI. 2, calculate the TI dataset in a backward manner. 3, customize the output filename. And update the alchemical_analysis.py parse_amber.py to use those options. Also I update the README file, basically just copy the information generated by "python alchemical_analysis.py --help" and insert into the usage section of the README file. The original README file is kind of out of data ie some changes of the option before my commit didn't reflect in the original README file.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -53,24 +53,33 @@ Help for `alchemical_analysis.py` (obtained with `alchemical_analysis -h`) is:
   -h, --help            show this help message and exit
   -a SOFTWARE, --software=SOFTWARE
                         Package's name the data files come from: Gromacs,
-                        Sire, or AMBER. Default: Gromacs.
+                        Sire, Desmond, or AMBER. Default: Gromacs.
+  -b BREAKS, --break=BREAKS
```
Comment (davidlmobley) [inline README.md]: I don't understand the "breaks" option -- neither why it's called "breaks" nor what the doc string means. Provide a bit more detail and a usage example, maybe? Also, for mnemonic value, the doc string should at least use the word "break" or "breaks".

Comment (davidlmobley) [reply README.md]: (I'm assuming this is some sort of a fraction since the default is 1.0, but it's very unclear.)

File: README.md [hunk 2]
```diff
@@ -53,24 +53,33 @@ Help for `alchemical_analysis.py` (obtained with `alchemical_analysis -h`) is:
...
+                        Extract the portion of the output files and calculate
+                        the data based on that portion. Default: 1.0
   -c, --cfm             The Curve-Fitting-Method-based consistency inspector.
                         Default: False.
   -d DATAFILE_DIRECTORY, --dir=DATAFILE_DIRECTORY
                         Directory in which data files are stored. Default:
                         Current directory.
+  -e, --backward        Using the backward direction for breaking the output
```
Comment (davidlmobley) [inline README.md]: I don't understand this doc string either. What's "backwardi" and how would one break the output file in teh backward direction?

File: alchemical_analysis/parser_amber.py [hunk 3]
```diff
@@ -624,6 +624,7 @@ def readDataAmber(P):
     # FIXME: compute maximum number of MBAR energy sections
     K = len(lvals)
     nsnapshots = [len(dvdl_all[clambda]) - start_from for clambda in lvals]
+    nsnapshots = [int(i*P.breaks) for i in nsnapshots]
```
Comment (davidlmobley) [inline alchemical_analysis/parser_amber.py]: Won't this go out of range in some cases?

**Comment by davidlmobley** [comment]
Looks mostly OK, except that the documentation needs improving as I can't understand some of these options in the README and help strings, @shuail .

File: README.md (hunk 1 above)
Comment (shuail) [reply README.md]: @davidlmobley initially I am thinking to use break to match -b as they all start with letter b. Yes, it's the fraction and I change the option word to fraction and update the doc string.

File: README.md (hunk 2 above)
Comment (shuail) [reply README.md]: @davidlmobley if using this option, the energy data will be extracted from the output file using the backward direction, yes, it's not break the output file but just extract them. I made a change to this doc string.

File: alchemical_analysis/parser_amber.py (hunk 3 above)
Comment (shuail) [reply alchemical_analysis/parser_amber.py]: @davidlmobley I don't think so, basically, it just update the nsnapthots by multiplying the numbers in the current nsnapshots list with the P.breaks (not P.fraction) value.

**Pull request merged**